src/robot/utils/robottime.py

```diff
@@ -316,6 +318,13 @@ def get_timestamp(daysep='', daytimesep=' ', timesep=':', millissep='.'):
     return TIMESTAMP_CACHE.get_timestamp(daysep, daytimesep, timesep, millissep)
 
 
+def get_timestamp_for_doc():
+    """Return a timestamp that honors `SOURCE_DATE_EPOCH`."""
+    ts = float(os.getenv('SOURCE_DATE_EPOCH', time.time()))
+    dt = datetime.datetime.fromtimestamp(round(ts), datetime.timezone.utc)
+    return dt.isoformat()
+
+
```

Comment:
This utility is used only in two places and only by Libdoc. I don't think adding it to `robot.utils` is a good idea because it's part of our public API and we cannot change it in the future without backwards compatibility concerns. There's no good place to have utils in the `robot.libdocpkg` package, but I think it would be fine to just inline these few lines of code. It would then make it possible to have slightly different formatting in the timestamp as well.

Reply:
Since the timestamp gets output in various files, you can't change it at any time lightly anyway, so I don't see the concern to have the procedure generating it live in a public API place.  As both a maintainer and user, I'd prefer consistency in all dates (figure its format once, and expect it everywhere).

Reply:
Actually, there could probably be a `timestamp` property in the `LibraryDoc` model object itself containing this functionality. That would avoid duplication.